[FIX] _parse_aeat_vat_info: ormcache and return correct value

Shide opened the pull request:
**Ahora el ormcache detecta el cambio de `aeat_identification_type`:**
Antes no lo detectaba y había que reiniciar el servidor para que cogiera el cambio. 
Cambiabas el `aeat_identification_type` de un cliente, ibas al libro de iva por ejemplo y al darle a recalcular, te cogía los datos anteriores.

**Devolver el `vat_number` y `identifier_type` cuando están los dos puestos:**
Como no se borra el campo de `aeat_identification` cuando cambias el `aeat_identification_type` en el partner, devolvía datos incorrectos.
Ahora si están puestos los dos, utiliza ese sistema, si no cogerá los datos del anterior proces (la lógica anterior)

Una vez fusionado haré cherry pick a la v15.

MT-706 @moduon

@pedrobaeza @rafaelbn @HaraldPanten

blaskurain committed:
[ADD] l10n_es_ticketbai_api_batuz, l10n_es_ticketbai_api Add Batuz support.

oca-travis committed:
[UPD] Update l10n_es_ticketbai_batuz.pot

OCA-git-bot committed:
[ADD] icon.png

enriquemartin committed:
[IMP] l10n_es_ticketbai_*batuz - nuevo modelo capítulo y gestión de capítulo 2

oca-travis committed:
[UPD] Update l10n_es_ticketbai_batuz.pot

OCA-git-bot committed:
l10n_es_ticketbai_batuz 11.0.2.0.0

enriquemartin committed:
[FIX] l10n_es_ticketbai_batuz - cálculo de clave CuotaIVADeducible en capítulo 2

oca-travis committed:
[UPD] Update l10n_es_ticketbai_batuz.pot

OCA-git-bot committed:
l10n_es_ticketbai_batuz 11.0.2.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-11.0/l10n-spain-11.0-l10n_es_ticketbai_batuz
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-11-0/l10n-spain-11-0-l10n_es_ticketbai_batuz/

ao-landoo committed:
[FIX] l10n_es_ticketbai_batuz: no dejar validar facturas rectificativas de proveedor sin factura rectificada asociada

oca-travis committed:
[UPD] Update l10n_es_ticketbai_batuz.pot

OCA-git-bot committed:
l10n_es_ticketbai_batuz 11.0.2.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-11.0/l10n-spain-11.0-l10n_es_ticketbai_batuz
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-11-0/l10n-spain-11-0-l10n_es_ticketbai_batuz/

ao-landoo committed:
[FIX] l10n_es_ticketbai_batuz: Fix multi-currency rate selection depending on date

oca-travis committed:
[UPD] Update l10n_es_ticketbai_batuz.pot

OCA-git-bot committed:
l10n_es_ticketbai_batuz 11.0.2.1.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-11.0/l10n-spain-11.0-l10n_es_ticketbai_batuz
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-11-0/l10n-spain-11-0-l10n_es_ticketbai_batuz/

enriquemartin committed:
[IMP] l10n_es_ticketbai - añadir tbai_active_date al diario para controlar la fecha a partir de la cual realizar comunicaciones con hacienda

enriquemartin committed:
[IMP] l10n_es_ticketbai - añadir tbai_active_date al diario para controlar la fecha a partir de la cual realizar comunicaciones con hacienda

oca-travis committed:
[UPD] Update l10n_es_ticketbai_batuz.pot

OCA-git-bot committed:
l10n_es_ticketbai_batuz 11.0.2.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-11.0/l10n-spain-11.0-l10n_es_ticketbai_batuz
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-11-0/l10n-spain-11-0-l10n_es_ticketbai_batuz/

enriquemartin committed:
[FIX] l10n_es_ticketbai_batuz - corregida clave régimen IVA en compras extracomunitarias

jesusVMayor committed:
[MIG]l10n_es_ticketbai_batuz: Migration to 12.0

jesusVMayor committed:
[FIX]l10n_es_ticketbai_batuz: Se buscan los impuestos de la compañía de la factura en vez de los del usuario.

oca-travis committed:
[UPD] Update l10n_es_ticketbai_batuz.pot

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update l10n_es_ticketbai_batuz.pot

OCA-git-bot committed:
l10n_es_ticketbai_batuz 12.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_ticketbai_batuz
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_ticketbai_batuz/

ao-landoo committed:
[IMP] l10n_es_ticketbai_batuz: black, isort, prettier

ao-landoo committed:
[MIG] l10n_es_ticketbai_batuz: Migration to 13.0

Commit:
[UPD] Update l10n_es_ticketbai_batuz.pot

OCA-git-bot committed:
[UPD] README.rst

ao-landoo committed:
[MIG] l10n_es_ticketbai_batuz: Migration to 14.0

OCA-git-bot committed:
delivery_gls_asm 14.0.1.4.3

OCA-git-bot committed:
[UPD] addons table in README.md

Shide committed:
[FIX] l10n_es_aeat: _parse_aeat_vat_info ormcache and getting identifier and vat number

- ormcache now detects when aeat_identification_type changes
- return the correct vat_number and identifier_type (when both are set)

[FIX] l10n_es_aeat: Tests for aeat_identification_type and aeat_identification

[FIX] l10n_es_aeat: add aeat_identification to ormcache, only check aeat_identification

- Adds aeat_identifcation to ormcache
- Only check aeat_identification to choose different identifier_type and vat_number

[FIX] l10n_es_aeat: check for aeat_identification_type instead of aeat_identification

jesusVMayor committed:
[FIX]l10n_es_ticketbai: Corregido filtered para evitar enviar facturas rectificativas cuando ticketbai no está habilitado.

jesusVMayor committed:
[FIX]l10n_es_ticketbai_batuz: Corregido dominio required reversed_entry_id.

pedrobaeza commented on l10n_es_aeat/models/res_partner.py:
```diff
@@ -72,8 +73,9 @@ def _parse_aeat_vat_info(self):
-            self.aeat_identification_type or identifier_type,
-            self.aeat_identification if self.aeat_identification_type else vat_number,
-        )
+
```
Por favor no introduzcas líneas vacías en mitad de un método.

pedrobaeza commented on l10n_es_aeat/models/res_partner.py:
```diff
@@ -72,8 +73,9 @@ def _parse_aeat_vat_info(self):
-            self.aeat_identification if self.aeat_identification_type else vat_number,
-        )
+
+        if self.aeat_identification_type and self.aeat_identification:
```
Está por pantalla como requerido el segundo campo, por lo que esto es redundante. Si por código se establece de forma incorrecta (uno establecido y el otro no), es responsabilidad de ese código el hacerlo bien, ¿no crees?

Shide replied:
Entiendo, quizás una mejor aproximación sería vaciar el campo de `self.aeat_identification` cuando se marque como vacío el `self.aeat_identification_type`. ¿Te parece que lo aborde así?

La cosa es que el cliente, al vaciar el campo de `self.aeat_identification_type`, puede entender que ya no se utiliza y que por lo tanto, el campo de `self.aeat_identification`, no importa si está rellenado o vacío.

pedrobaeza replied:
Si `_parse_aeat_vat_info` considera el campo `aeat_identification_type` para ya leer el siguiente campo (como estaba antes), realmente no necesitaríamos tener que vaciar el campo, y simplificamos la lógica.

Shide replied:
Entonces con un `if self.aeat_identification_type:`, sería suficiente.

pedrobaeza replied:
Sí, creo que es lo mejor si no se ve ninguna contraindicación.

pedrobaeza reviewed (commented):
Lo de la caché lo veo bien, e incluso añadir también `self.aeat_identification`, pero los otros cambios no los veo necesarios.

pedrobaeza commented:
Además, el mensaje de commit debe contener el nombre del módulo: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#71commit-message

OCA-git-bot committed:
l10n_es_ticketbai_batuz 13.0.1.1.1

OCA-git-bot committed:
l10n_es_ticketbai 14.0.2.6.4

OCA-git-bot committed:
[UPD] addons table in README.md

Commit:
[14.0][IMP] l10n_es_ticketbai_batuz: black, isort, prettier

victoralmau committed:
[IMP] l10n_es_aeat_sii_oca: Avoid modifying third-party number when an invoice has already been sent to the SII + Tests.
[FIX] l10n_es_aeat_sii_oca: Fix incorrect invoice type (in refund) to in_refund
TT38248

Commit:
[UPD] Update l10n_es_aeat_sii_oca.pot

Commit:
[UPD] Update l10n_es_ticketbai_batuz.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

Commit:
[UPD] Update l10n_es_aeat_sii_oca.pot

OCA-git-bot committed:
l10n_es_aeat_sii_oca 14.0.2.3.1

OCA-git-bot committed:
[UPD] addons table in README.md

Shide commented:
se me ha ido la mano... voy a abrir otra rama... sorry...

pedrobaeza commented:
Bueno, no es necesario haciendo `push -f` si no quieres, pero lo que prefieras.

Shide commented:
Perdón @pedrobaeza, lo he hecho en otra rama... https://github.com/OCA/l10n-spain/pull/2455
